src/NovelRunner.cpp

```diff
@@ -7,22 +7,27 @@
 #include <SDL2/SDL.h>
 
 namespace NovelRT {
-NovelRunner::NovelRunner(int displayNumber, NovelLayeringService* layeringService)
-    : _layeringService(layeringService), _novelRenderer(std::make_unique<NovelRenderingService>(_layeringService)) {
+NovelRunner::NovelRunner(int displayNumber, NovelLayeringService* layeringService, uint32_t targetFrameRate)
+    : _layeringService(layeringService), _novelRenderer(std::make_unique<NovelRenderingService>(_layeringService)), _stepTimer(targetFrameRate) {
   _novelRenderer->initialiseRendering(displayNumber);
   _novelInteractionService = std::make_unique<NovelInteractionService>(_layeringService, _novelRenderer->getScreenScale());
   _novelInteractionService->subscribeToQuit([this]{_exitCode = 0;});
 }
 
 int NovelRunner::runNovel() {
-  Uint64 current = SDL_GetPerformanceCounter();
-  Uint64 previous = 0;
-  float deltaTime = 0;
+  uint32_t lastFramesPerSecond = 0;
+
   while (_exitCode) {
-    previous = current;
-    current = SDL_GetPerformanceCounter();
-    deltaTime = ((current - previous) * 1000 / SDL_GetPerformanceFrequency()) * 0.001f;
-    executeUpdateSubscriptions(deltaTime);
+    _stepTimer.tick(_updateSubscribers);
+
+    auto framesPerSecond = _stepTimer.getFramesPerSecond();
+
+    if (framesPerSecond != lastFramesPerSecond) {
```

Comment:
This should be replaced once the support for rendering text elements is merged

Reply:
What would you want to replace it with?